niworkflows/interfaces/bids.py

```diff
@@ -240,7 +240,10 @@ class _DerivativesDataSinkInputSpec(DynamicTraitedSpec, BaseInterfaceInputSpec):
     keep_dtype = traits.Bool(False, usedefault=True, desc='keep datatype suffix')
     meta_dict = traits.DictStrAny(desc='an input dictionary containing metadata')
     source_file = File(exists=False, mandatory=True, desc='the input func file')
-    space = Str('', usedefault=True, desc='Label for space field')
+    space = traits.Either(Str, traits.Tuple(Str, traits.Dict),
+                          default='', usedefault=True,
+                          desc='Label for space field. Accepts a template string or a '
+                               'tuple in the form of (template, {template specifications})')
```

Comment:
Does it make sense to make `res` and `den` subordinate to `space`? Conceptually, that makes sense, but in terms of generic BIDS entities, maybe it would make more sense to abstract into `entities`, which would allow any additional entities to be specified.

But possibly that's too much of a pain. IDK. We might just need to revisit this whole interface sooner or later.

Reply:
Yeah, I guess in a BIDS context `space`, `res`, and `den` are all separate entities, so this can be a little confusing. Though they are closely related in an internal fmriprep aspect (the upstream PR adds support for multiple configurations of the same space).

I'll try playing around with the interface's `allowed_entities` functionality to see if I can get the desired behavior without altering this.

Reply:
actually, i don't think this is possible using `allowed_entities`, since the dictionary containing the space specifications is unable to be passed through workflow connections.

or were you suggesting adding a new input `entities` that would handle extra fields?